src/middlewared/middlewared/plugins/cloud_sync.py

```diff
@@ -611,6 +611,9 @@ async def do_delete(self, id):
         """
         Delete Cloud Sync Credentials of `id`.
         """
+        if await self.middleware.call("cloudsync.query", [["credential", "=", id]]):
+            raise CallError("This credential is used")
```

Comment:
Should we use errno.EBUSY?

@Mrt134 any thoughts in the error message?

Reply:
@Mrt134 ping

Reply:
Is there any way to specify which cloud task the credential is being used by?

Reply:
That way you could say something like `This credential is in use by: blah`. If not, then the verbiage you have is fine.